### DISCUSSION TITLE
Static website with restricted access through VPN

### DISCUSSION THREAD

**Discussion by lightmagic1** [vercel/vercel#5847] [Help]
Hello everyone!

I've been searching more than a week a good way to do this:

Host a static website only accessible through VPN.

My pipeline generates a build folder with all static assets (html, css, js), i should upload it and only people within the following ip ranges access my website on mydomain.com

172.3.X.X/14
172.4.X.X/14

Can i do something like that using vercel?

**Comment by paulogdm** [comment]
It is not possible to restrict Vercel sites that way. However, you can enable password protection and other security features.

**Reply by mwalden2004** [reply]
Ah yes, with a $150 payment per month. What a simple feature to make you pay so much.

**Comment by logantai24** [comment]
@lightmagic1  I think you can do it together with **Cloudflare for Teams** (free for up to 50 members). 

1. Use Cloudflare to proxy your vercel site 
https://vercel.com/support/articles/using-cloudflare-with-vercel 

2. Add your site to your Cloudflare for Teams account as an application
https://developers.cloudflare.com/cloudflare-one/applications/configure-apps/self-hosted-apps/

3. Set up policies that only allow access within certain IP ranges

**Reply by leog** [reply]
Would that mean that only different 50 users can access the website through that range of IPs? Or the 50 users restriction is just the members allowed to access Cloudflare dashboard?

**Reply by logantai24** [reply]
@leog Only different 50 users can access the website through that range of IPs, once exceeding 50 users, Cloudflare will charge you for $7/per user
